spring-integration-core/src/main/java/org/springframework/integration/transaction/DefaultTransactionSynchronizationFactory.java

```diff
@@ -46,13 +49,16 @@ public DefaultTransactionSynchronizationFactory(TransactionSynchronizationProces
 	@Override
 	public TransactionSynchronization create(Object key) {
 		Assert.notNull(key, "'key' must not be null");
-		DefaultTransactionalResourceSynchronization synchronization = new DefaultTransactionalResourceSynchronization(key);
-		TransactionSynchronizationManager.bindResource(key, synchronization.getResourceHolder());
-		return synchronization;
+		if (!TransactionSynchronizationManager.hasResource(key)) {
```

Comment:
We seem to have broken this "factory" in INT-3408; it's no longer a simple factory that creates a sync. It's hard to see how a user TSF that does __not__ bind the resource will work properly - the recent code to provide the message after a commit failure will not work if it doesn't do that. Hence I believe the caller (APE etc) __must__ do the binding.